typescript/web/src/components/pagination-tool/pagination-tool.tsx

```diff
@@ -0,0 +1,134 @@
+import {
+  // useColorModeValue as mode,
+  HStack,
+  IconButton,
+  NumberInput,
+  NumberInputField,
+  Tooltip,
+  Text,
+} from "@chakra-ui/react";
+import { RiArrowLeftSLine, RiArrowRightSLine } from "react-icons/ri";
+import { isNaN, clamp } from "lodash/fp";
+import { useEffect, useState } from "react";
+
+interface PaginationToolProps {
+  currentPage: number;
+  maxPage: number;
+  onChange: (newPage: number) => void;
+}
+
+const digitsPerRem = 0.55;
+
+export const PaginationTool = (props: PaginationToolProps) => {
+  const { currentPage, maxPage, onChange: setPage } = props;
+  const [inputValue, setInputValue] = useState(currentPage);
+  const digitCount = Math.max(1, Math.ceil(Math.log10(maxPage ?? 1)));
+
+  useEffect(() => {
+    setInputValue(currentPage);
+  }, [currentPage]);
+
+  const handleKeyPress = (event: any) => {
+    if (event.key === "Enter") {
+      setPage(event.target.value as number);
+    }
+    if (event.key === "Escape" || event.key === "Esc") {
+      event.preventDefault();
+      setInputValue(currentPage);
+    }
+  };
+
+  return (
+    <HStack h={10} p={0} spacing={1} rounded={6} pointerEvents="initial">
+      {currentPage > 1 ? (
+        <Tooltip
+          openDelay={300}
+          label={`Previous Page: ${currentPage - 1}`}
+          placement="top"
+        >
+          <IconButton
+            bg="none"
+            aria-label="Previous page"
+            icon={<RiArrowLeftSLine size="1.5em" />}
+            onClick={() => setPage(currentPage - 1)}
+          />
+        </Tooltip>
+      ) : (
+        <IconButton
+          disabled
+          bg="none"
+          aria-label="No previous page"
+          icon={<RiArrowLeftSLine size="1.5em" />}
+        />
+      )}
+      <Tooltip label="Current page" placement="top" openDelay={300}>
+        <NumberInput
+          name="current-page"
+          rounded={6}
+          allowMouseWheel
+          defaultValue={currentPage != null ? currentPage : 1}
+          min={1}
+          max={maxPage}
+          variant="filled"
+          textAlign="center"
+          size="sm"
+          value={inputValue}
+          onChange={(valStr, valNum) => {
+            if (!isNaN(valNum)) {
+              setInputValue(clamp(valNum, 1, maxPage));
+            }
+          }}
+          onKeyPress={handleKeyPress}
+          onBlur={() => {
+            setPage(inputValue);
+          }}
+        >
+          <NumberInputField
+            rounded={6}
+            onClick={(event) => {
+              (event.target as HTMLInputElement).select();
+            }}
+            textAlign="center"
+            w={`${digitCount * digitsPerRem + 1.5}rem`}
```

Comment:
I'd use here a Chakra value for the width, such as `2`, `3` or `4`. This enables to have a consistent design across the different components (same thing below)

Reply:
I have taken this code from the component used for navigating between image in the labelling view, it allows for scaling the number input field to fit the same number of digits as the maximum page number

Reply:
@crubier do you have an answer to that? It seems that you originally coded that part on the [image navigation tool](https://github.com/labelflow/labelflow/blob/752c7389a11e32af6c4086facc2a97e4c0c4c858/typescript/web-app/src/components/image-navigation-tool-bar/image-navigation-tool.tsx), which I partially copied

Reply:
This number only depends on the font family used. We define the font in CSS so there is no reason for this number to change. It does not depend on zoom or font scaling since it is expressed in rem which also scale with zoom or font scaling, canceling each other. 

I found this number by carefully trying various values in storybook, and finding the optimal value that: 

- is not too wide, to avoid having an overly large margin in the box when the number is made of narrow digits (worst case is 11111)
- is not too narrow, to avoid cropping/overflowing the box when the number is made of wide digits (worst case is 88888 or 66666 i think)

The value is precise and 0.6 is too large and 0.5 is too small. 

I don’t think there is an easy way to test that, and anyway you’d be testing the test engine font rendering more than your own code. Also this value is not likely to change often so there is no reason to spend energy testing it really.

Reply:
There are other methods of fitting an input box size to its content but 

- it’s better to only depend on the number of digits and not on the actual exact width of the number, to avoid constantly changing the width of the box every time the number change, which would be bad UX  (here we only change the width when we go from 9 to 10, … , 9999 to 10000 etc..)
- the methods to fit an input to its content often involve complex measurement process that involve DOM operations and CSS which are too costly for the need